pkgs/by-name/ha/handheld-daemon/package.nix

```diff
@@ -1,63 +1,96 @@
 {
+  lib,
+  python3Packages,
   fetchFromGitHub,
+
+  # dependencies
+  systemd,
   hidapi,
-  kmod,
-  lib,
-  python3,
   toybox,
-  lsof
+  kmod,
+  dbus,
+  lsof,
 }:
-python3.pkgs.buildPythonApplication rec {
+python3Packages.buildPythonApplication rec {
   pname = "handheld-daemon";
-  version = "3.6.1";
+  version = "3.6.2";
   pyproject = true;
 
   src = fetchFromGitHub {
     owner = "hhd-dev";
     repo = "hhd";
     rev = "refs/tags/v${version}";
-    hash = "sha256-IdpSRb66G+WzTv/BC29r2OjO1b4VdWbV6OSzOoiFAO0=";
+    hash = "sha256-W1Ap6yTryBDozKe3aO413Fu0RBul9kEA9ACUTdYyOKM=";
   };
 
-  propagatedBuildInputs = with python3.pkgs; [
+  postPatch =
+    let
+      udevadm = lib.getExe' systemd "udevadm";
```

Comment:
I know you're doing this to avoid repeating yourself, but I suspect being congruent with the other calls is more valuable than saving two lookups.

I suggest either:

- inline `udevadm` like the other dependencies, or
- make variables for all the dependencies like `chmodPath = lib.getExe' toybox "chmod"` and use them (`${ chmodPath }`) for all substitutions below.

Reply:
Yes, fair. I inlined everything.